addons/position_manager/position_manager.lua

```diff
@@ -0,0 +1,83 @@
+--Copyright (c) 2020, Lili
+--All rights reserved.
+
+--Redistribution and use in source and binary forms, with or without
+--modification, are permitted provided that the following conditions are met:
+
+--    * Redistributions of source code must retain the above copyright
+--      notice, this list of conditions and the following disclaimer.
+--    * Redistributions in binary form must reproduce the above copyright
+--      notice, this list of conditions and the following disclaimer in the
+--      documentation and/or other materials provided with the distribution.
+--    * Neither the name of position_manager nor the
+--      names of its contributors may be used to endorse or promote products
+--      derived from this software without specific prior written permission.
+
+--THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+--ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+--WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+--DISCLAIMED. IN NO EVENT SHALL Lili BE LIABLE FOR ANY
+--DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+--(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+--LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+--ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+--(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+--SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+_addon.name = 'position_manager'
+_addon.author = 'Lili'
+_addon.version = '1.0.0'
+_addon.command = 'pm'
+
+if not windower.file_exists(windower.windower_path .. '\\plugins\\WinControl.dll') then
+    print('position_manager: error - Please install the WinControl plugin in the launcher.')
+    windower.send_command('lua u position_manager')
+    return
+end
+
+config = require 'config'
+
+default = {
+    x = 0,
+    y = 0,
+}
+
+settings = config.load(default)
+
+function move(settings)
+    windower.send_command('load wincontrol')
+    coroutine.sleep(0.5)
+    windower.send_command('wincontrol move %s %s':format(settings.x,settings.y))
+    coroutine.sleep(0.5)
+    windower.send_command('unload wincontrol')
+end
+
+function handle_commands(cmd,name,pos_x,pos_y)
+    local cmd = cmd or 'help'
+    
+    if cmd == 'r' then
+        windower.send_command('lua r position_manager')
+    elseif cmd == 'set' then
+        if name and pos_x and pos_y then
+            settings.x = tonumber(pos_x)
+            settings.y = tonumber(pos_y)
+            config.save(settings,name)
+            
+            if windower.ffxi.get_info().logged_in then
+                player_name = windower.ffxi.get_player().name
+                if name:lower() == player_name:lower() then
+                    move(settings)
+                end
+                -- TODO: add IPC
```

Comment:
This is something I am very unsure about. It will make moving the *current* window a chore, since you always have to look up and type out the character name. How about foregoing IPC completely and only make it work locally? It could be that this is me being out of touch with the game, is it so uncommon to switch to a mule's window? I know I wanted to control my mule without ever having to switch to their screen, but that was about gameplay, setting up window positions once in a while is a situation where I feel that having to switch to the respective window is a reasonable expectation.

And do not misunderstand me, I do feel this is a useful feature, but it does come at a cost, namely making adjusting the *current* window harder to adjust.

Reply:
This is 100% designed around my own use case, where I was able to set a position for all of my characters while sitting in front of my main, _including characters that weren't logged in_ (i.e. same-account alts). 

What do you think about a `current` keyword instead? The addon would sub in the current character's name. 
Or I could make the name parameter optional entirely, altough I'm aware that you don't like that too much :P

Reply:
Good point, I had not considered logged out characters. That could be useful.

In that case I would suggest switching the order, append the character name, or omit it if you want the current character. Does that sound ok? I would not min the current solution either by the way, it was just something that I thought could be streamlined.

Reply:
Switched the order around, changed `all` to `:all` because magic word, added some sanity checks. Gave up on `:current` as it'd be redundant with the order switched.